src/core/or/circuitpadding.h

```diff
@@ -0,0 +1,539 @@
+/*
+ * Copyright (c) 2017, The Tor Project, Inc. */
+/* See LICENSE for licensing information */
+
+/**
+ * \file circuitpadding.h
+ * \brief Header file for circuitpadding.c.
+ **/
+#ifndef TOR_CIRCUITPADDING_H
+#define TOR_CIRCUITPADDING_H
+
+#include "circpad_negotiation.h"
+#include "lib/evloop/timers.h"
+
+typedef struct circuit_t circuit_t;
+typedef struct origin_circuit_t origin_circuit_t;
+typedef struct cell_t cell_t;
+
+/**
+ * Circpad state specifier.
+ *
+ * Each circuit has up to two state machines, and each state
+ * machine consists of these states. Machines transition between
+ * these states using the event transition specifiers below.
+ */
+typedef enum {
+  CIRCPAD_STATE_START = 0,
+  CIRCPAD_STATE_BURST = 1,
+  CIRCPAD_STATE_GAP = 2,
+  CIRCPAD_STATE_END = 3
+} circpad_statenum_t;
+#define CIRCPAD_NUM_STATES  ((uint8_t)CIRCPAD_STATE_END+1)
+
+/**
+ * These constants form a bitfield to specify the types of events
+ * that can cause transitions between state machine states.
+ *
+ * Note that SENT and RECV are relative to this endpoint. For
+ * relays, SENT means packets destined towards the client and
+ * RECV means packets destined towards the relay. On the client,
+ * SENT means packets destined towards the relay, where as RECV
+ * means packets destined towards the client.
+ */
+typedef enum {
+  CIRCPAD_EVENT_NONPADDING_RECV = 1<<0,
+  CIRCPAD_EVENT_NONPADDING_SENT = 1<<1,
+  CIRCPAD_EVENT_PADDING_SENT = 1<<2,
+  CIRCPAD_EVENT_PADDING_RECV = 1<<3,
+  CIRCPAD_EVENT_INFINITY = 1<<4,
```

Comment:
Infinity of what?  I think we should have documentation for each of these enum values.

Reply:
Addressed in 80ab22476ff2c7e2a6ce91f74879aec18b81a9bc.